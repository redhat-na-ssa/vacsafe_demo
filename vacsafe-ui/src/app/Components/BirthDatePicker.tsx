import { 
  americanDateFormat, 
  birthDateValidator, 
  birthDateRules 
} from '@app/utils/utils';
import { 
  FormGroup, 
  DatePicker 
} from '@patternfly/react-core';
import * as React from 'react';
import { Controller } from 'react-hook-form';

const BirthDatePicker = ({control, defaultBirthDate}) => {
  return (
    <Controller
      name="dateOfBirth"
      control={control}
      defaultValue={defaultBirthDate}
      render={({ field, fieldState }) => (
        <FormGroup
          label="Date of Birth"
          isRequired
          helperTextInvalid="This field is required. Please choose your date of birth in MM-DD-YYYY format."
          fieldId="date-of-birth"
          validated={fieldState.error ? "error" : "default"}
        >
          <DatePicker
            aria-label="date-of-birth"
            value={field.value}
            dateFormat={date => date.toLocaleDateString('en-US', { year: 'numeric', month: '2-digit', day: '2-digit' }).replace(/\//g, '-')}
            dateParse={americanDateFormat}
            placeholder="MM-DD-YYYY"
            helperText="Please pick your birth date."
            onChange={field.onChange}
            validators={[birthDateValidator]}
          />
        </FormGroup>
      )}
      rules={{ required: true, validate: birthDateRules, pattern: /^(0[1-9]|1[0-2])\-(0[1-9]|1\d|2\d|3[01])\-(19|20)\d{2}$/ }}
    />
  )
}

export default BirthDatePicker; 