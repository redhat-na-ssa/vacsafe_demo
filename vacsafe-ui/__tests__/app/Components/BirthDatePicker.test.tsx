import React from 'react';
import { shallow } from 'enzyme';
import BirthDatePicker from '../../../src/app/Components/BirthDatePicker';

describe('<BirthDatePicker />', ()=> {
    it('should be defined', () => {
        expect(BirthDatePicker).toBeDefined()
    })
    it("should render without crashing", () => {
        const wrapper = shallow(<BirthDatePicker control={undefined} defaultBirthDate={undefined} />)
        expect(wrapper)
    })
    it("should match existing snapshot", () => {
        const wrapper = shallow(<BirthDatePicker control={undefined} defaultBirthDate={undefined} />)
        expect(wrapper).toMatchSnapshot()
    })
})
