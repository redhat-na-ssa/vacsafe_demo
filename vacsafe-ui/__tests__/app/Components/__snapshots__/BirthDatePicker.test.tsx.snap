// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BirthDatePicker /> should match existing snapshot 1`] = `
<Component
  name="dateOfBirth"
  render={[Function]}
  rules={
    Object {
      "pattern": /\\^\\(0\\[1-9\\]\\|1\\[0-2\\]\\)\\\\-\\(0\\[1-9\\]\\|1\\\\d\\|2\\\\d\\|3\\[01\\]\\)\\\\-\\(19\\|20\\)\\\\d\\{2\\}\\$/,
      "required": true,
      "validate": [Function],
    }
  }
/>
`;
